homeassistant/helpers/device_registry.py

```diff
@@ -489,6 +494,8 @@ async def async_load(self) -> None:
                     connections={tuple(conn) for conn in device["connections"]},  # type: ignore[misc]
                     identifiers={tuple(iden) for iden in device["identifiers"]},  # type: ignore[misc]
                     id=device["id"],
+                    # Introduced in 2021.2
+                    orphaned_timestamp=device.get("orphaned_timestamp"),
```

Comment:
`time.time()` used here instead of a `datetime` object since the database backend we are using here is a `json` file and we have to pay the overhead serialize and de-serialize the data each time which is done frequently. Conversion of to and from `datetime` objects turned out to be quite expensive with a lot of deleted devices, and considering this file isn't meant to be human consumable, it didn't seem worth it.